crates/api_crud/src/site/read.rs

```diff
@@ -99,8 +107,31 @@ pub async fn get_site(
       discussion_languages,
     })
   } else {
+    // filter oauth_providers for public access
+    filter_oauth_providers(&mut site_response.oauth_providers);
     None
   };
 
   Ok(Json(site_response))
 }
+
+fn filter_oauth_providers(oauth_providers: &mut Vec<OAuthProvider>) {
+  oauth_providers.retain_mut(|oauth_provider| {
+    if oauth_provider.enabled.unwrap_or(false) {
+      oauth_provider.issuer = None;
+      oauth_provider.token_endpoint = None;
+      oauth_provider.userinfo_endpoint = None;
+      oauth_provider.id_claim = None;
+      oauth_provider.name_claim = None;
+      oauth_provider.auto_verify_email = None;
+      oauth_provider.auto_approve_application = None;
+      oauth_provider.account_linking_enabled = None;
+      oauth_provider.enabled = None;
+      oauth_provider.published = None;
+      oauth_provider.updated = None;
+      true
+    } else {
+      false
```

Comment:
Something tells me there's a better way to organize this. Like maybe it should be added as an impl function on the `OAuthProvider` struct.

Reply:
The whole filter function, but I wouldn't worry about it for now, I can always fix later.